tmt/base.py

```diff
@@ -1032,6 +1042,151 @@ def go(self):
+        else:
+            return style(content, fg='yellow')
+
+    @classmethod
+    def pad_with_spaces(cls, string):
+        """ Append spaces to string to properly align the first column """
+        return string + (cls.FIRST_COL_LEN - len(string)) * ' '
+
+    def run_matches_filters(self, run):
+        """ Check if the given run matches filters from the command line """
+        if self.opt('abandoned') or self.opt('active'):
+            # Any of the plans must be abandoned/active for the whole
+            # run to be abandoned/active
+            return any(self.plan_matches_filters(p) for p in run.plans)
+        if self.opt('finished'):
+            # All plans must be finished for the whole run to be finished
+            return all(self.plan_matches_filters(p) for p in run.plans)
+        return True
+
+    def print_run_status(self, run):
+        """ Display the overall status of the run """
+        if not self.run_matches_filters(run):
+            return
+        # Find the earliest step in all plans' status
+        earliest_step_index = len(tmt.steps.STEPS)
+        for plan in run.plans:
+            plan_status = self.get_overall_plan_status(plan)
+            if plan_status == 'done':
+                continue
+            elif plan_status == 'todo':
+                # If plan has no steps done, consider the whole run not done
+                earliest_step_index = -1
+                break
+            plan_status_index = tmt.steps.STEPS.index(plan_status)
+            if plan_status_index < earliest_step_index:
+                earliest_step_index = plan_status_index
+
+        if earliest_step_index == len(tmt.steps.STEPS):
+            run_status = 'done'
+        elif earliest_step_index == -1:
+            run_status = 'todo'
+        else:
+            run_status = tmt.steps.STEPS[earliest_step_index]
+        run_status = self.colorize_column(self.pad_with_spaces(run_status))
+        echo(run_status, nl=False)
+        echo(run.workdir)
+
+    def print_plans_status(self, run):
+        """ Display the status of each plan of the given run """
+        for plan in run.plans:
+            if self.plan_matches_filters(plan):
+                plan_status = self.get_overall_plan_status(plan)
+                echo(self.colorize_column(self.pad_with_spaces(plan_status)),
+                     nl=False)
+                echo(f'{run.workdir}  {plan.name}')
+
+    def print_verbose_status(self, run):
+        """ Display the status of each step of the given run """
+        for plan in run.plans:
+            if self.plan_matches_filters(plan):
+                for step in plan.steps(disabled=True):
+                    column = step.status() + ' '
+                    echo(self.colorize_column(column), nl=False)
+                echo(f' {run.workdir}  {plan.name}')
+
+    def process_run(self, run):
+        """ Display the status of the given run based on verbosity """
+        run.load()
```

Comment:
It would be good to handle possible `GeneralError` exception here becase the metadata tree root might not be always accessible. In that case the first such run causes this error:

    Invalid yaml syntax: Invalid directory path: /home/psss/git/tmt/tests/test/select/data

Reply:
Perhaps something like this?

```python
try:
    run.load()
except tmt.utils.GeneralError as error:
    self.warn(f'Failed to check {run.workdir} ({error}).')
    return
```